### PR TITLE
feat Add Endpoint to filter messages by valid date

### PR DESCRIPTION
**Pull request by Doug-Reed**

### CODE REVIEW

File: src/main/java/edu/wisc/my/messages/controller/MessagesController.java [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
+import edu.wisc.my.messages.service.MessagesService;
+
 @Controller
 public class MessagesController {
     protected final Logger logger = LoggerFactory.getLogger(getClass());
     private MessagesService messagesService;
     
     @RequestMapping(value="/messages", method=RequestMethod.GET)
-    public @ResponseBody void getJson(HttpServletRequest request,
+    public @ResponseBody void getAllMessages(HttpServletRequest request,
```
Comment (apetro) [inline src/main/java/edu/wisc/my/messages/controller/MessagesController.java]: Stipulating that I'm in the minority about this and you should all just tell me I'm wrong: 

I object to this method naming convention. This isn't a JavaBean accessor, it shouldn't be named with a `get` prefix. It's just `public @ResponseBody void messages(...)` or `...allMessages(...)` or so.

File: src/main/java/edu/wisc/my/messages/controller/MessagesController.java [hunk 2]
```diff
@@ -31,6 +35,33 @@
                 response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             }
         }
+
+    @RequestMapping(value="/datedMessages", method=RequestMethod.GET)
```
Comment (apetro) [inline src/main/java/edu/wisc/my/messages/controller/MessagesController.java]: Not sure `/datedMessages` is the most expressive way to name this web service. Maybe `/currentMessages`? `/messagesFilteredToCurrent` ?

Which maybe leads to an insight that this should be a nuance on `/messages`, as in taking a `filter` request param where a `toCurrent` value turns on the date filtering?

Should filtering by date be the exception? Maybe `/messages` should filter to current by default, and then there should be an exceptional `/allMessages` or `/messages?filtered=false` that doesn't filter that we might use in e.g. admin views?

File: src/main/java/edu/wisc/my/messages/controller/MessagesController.java [hunk 3]
```diff
@@ -31,6 +35,33 @@
                 response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             }
         }
+
+    @RequestMapping(value="/datedMessages", method=RequestMethod.GET)    
+    public @ResponseBody void getDatedMessages(HttpServletRequest request,
```
Comment (vertein) [inline src/main/java/edu/wisc/my/messages/controller/MessagesController.java]: Do we need a separate endpoint for all the message filtering, or should this just be a feature of all messages?

I think a lot of this (business logic) can be moved into the service and leave the controller just for the delivering content purpose.

File: src/main/java/edu/wisc/my/messages/controller/MessagesController.java [hunk 4]
```diff
@@ -31,6 +35,33 @@
...
+      HttpServletResponse response) {
+        response.setContentType("application/json");
+        JSONObject rawMessages = messagesService.getRawMessages();
+        JSONObject validMessages = new JSONObject();
```
Comment (apetro) [inline src/main/java/edu/wisc/my/messages/controller/MessagesController.java]: I don't think `valid` is the best way to characterize what we're filtering to here. Aren't they more `current` than they are `valid`?

File: src/main/java/edu/wisc/my/messages/model/Message.java [hunk 5]
```diff
@@ -291,6 +315,55 @@ public ActionButton getConfirmButton() {
...
+   Date today = new Date();
+   
+   // Placeholder date format string, 
+   // can be overridden with an application property.
+   String formatString = "yyyy-mm-dd";
+   
+   if(env == null){
+     logger.error("NO ENV VARIABLE");
+   } else {
+      formatString = env.getProperty("default.date.format");
```
Comment (apetro) [inline src/main/java/edu/wisc/my/messages/model/Message.java]: YAGNI? `yyyy-mm-dd` is the ISO format, let's just hard code to that format until a requirement comes along to do something different?

File: src/main/java/edu/wisc/my/messages/controller/MessagesController.java (hunk 1 above)
Comment (vertein) [reply src/main/java/edu/wisc/my/messages/controller/MessagesController.java]: I would vote messages(), but I'm not too strong on this

File: src/main/java/edu/wisc/my/messages/model/Message.java [hunk 6]
```diff
@@ -291,6 +315,55 @@ public ActionButton getConfirmButton() {
...
+  
+  public boolean isValidToday(){
+
+   Date today = new Date();
+   
+   // Placeholder date format string, 
+   // can be overridden with an application property.
+   String formatString = "yyyy-mm-dd";
+   
+   if(env == null){
```
Comment (davidmsibley) [inline src/main/java/edu/wisc/my/messages/model/Message.java]: Couple of things here:
1. `env` will be autowired by Spring, if `env` is null, you've got larger problems
2. I'm not sure if it's still the case, but usually, checking `env` is expensive. The default date format will never change during the runtime, so it'd be much more efficient to get the date format at static initialization (or at the very least instance instantiation) and then just reference that object from within this method.

**Pull request merged**